components/usage/pkg/apiv1/usage.go

```diff
@@ -181,6 +181,42 @@ func (s *UsageService) GetCostCenter(ctx context.Context, in *v1.GetCostCenterRe
 	}, nil
 }
 
+func (s *UsageService) ReconcileUsageWithLedger(ctx context.Context, req *v1.ReconcileUsageWithLedgerRequest) (*v1.ReconcileUsageWithLedgerResponse, error) {
+	from := req.GetFrom().AsTime()
+	to := req.GetTo().AsTime()
+
+	logger := log.
+		WithField("from", from).
+		WithField("to", to)
+
+	if to.Before(from) {
+		return nil, status.Errorf(codes.InvalidArgument, "To must not be before From")
+	}
+
+	stopped, err := db.FindStoppedWorkspaceInstancesInRange(ctx, s.conn, from, to)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to find stopped workspace instances.")
+		return nil, status.Errorf(codes.Internal, "failed to query for stopped instances")
+	}
+	logger.Infof("Found %d stopped workspace instances in range.", len(stopped))
+
+	running, err := db.FindRunningWorkspaceInstances(ctx, s.conn)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to find running workspace instances.")
+		return nil, status.Errorf(codes.Internal, "failed to query for running instances")
+	}
+	logger.Infof("Found %d running workspaces since the beginning of time.", len(running))
+
+	usageDrafts, err := db.FindAllDraftUsage(ctx, s.conn)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to find all draft usage records.")
+		return nil, status.Errorf(codes.Internal, "failed to find all draft usage records")
+	}
+	logger.Infof("Found %d draft usage records.", len(usageDrafts))
```

Comment:
We'll also need to fetch the instances for the drafts and then combine the three slices of instances. But that's part of the next iteration I guess?